src/Illuminate/Routing/UrlGenerator.php

```diff
@@ -239,9 +239,7 @@ public function asset($path, $secure = null)
         // Once we get the root URL, we will check to see if it contains an index.php
         // file in the paths. If it does, we will remove it since it is not needed
         // for asset paths, but only for routes to endpoints in the application.
-        $root = $this->assetRoot
-                    ? $this->assetRoot
-                    : $this->formatRoot($this->formatScheme($secure));
+        $root = $this->assetRoot ?: $this->formatRoot($this->formatScheme($secure));
```

Comment:
This is a wrong action.

If it comes as $this->assetRoot (if true)
? $this->assetRoot (if not empty)
: $this->formatRoot ($this->formatScheme($secure));

If both are true, $this->assetRoot is used. Otherwise, the third statement is used.

Your transaction is used even if the related $assetRoot is empty.

This situation may send incorrect results to the developer.